src/main/java/com/mraof/minestuck/tileentity/redstone/WirelessRedstoneTransmitterTileEntity.java

```diff
@@ -42,20 +46,42 @@ public void tick()
 		tickCycle++;
 	}
 	
-	public BlockPos getDestinationBlockPos()
+	public BlockPos getDestinationBlockPosFromOffset()
 	{
-		if(destBlockPos == null)
-			destBlockPos = new BlockPos(0, 0, 0);
-		return destBlockPos;
+		if(offsetPos == null)
+			offsetPos = new BlockPos(0, 0, 0);
+		if(facing == null)
+			facing = getBlockState().getValue(WirelessRedstoneTransmitterBlock.FACING);
+		
+		Direction stateFacing = getBlockState().getValue(WirelessRedstoneTransmitterBlock.FACING);
+		
+		//in cases where the block has been rotated after the coordinates were set(rotator blocks or placement in structure templates), this will transform the relative coordinates by rotating them around the block
+		if(facing != stateFacing)
+			this.offsetPos = offsetPos.rotate(MSRotationUtil.rotationBetween(this.facing, stateFacing));
```

Comment:
I'd suggest removing the need for a `facing` field by doing one of:
- Let the offset position always be the unrotated offset, and in this function get a rotation from `stateFacing` to them apply with `this.getBlockPos().offset(offsetPos.rotate(rotation))`
- Introduce functionality so that the rotator block can call some function to rotate `offsetPos`

Reply:
Do you think its an issue to store the facing?

Reply:
Not necessarily, and I do think what you've written works correctly. Rather, I think that the strategy with the cached facing makes the tile entity more complicated than it has to be for no clear benefit.
(I also see it as a bonus to change `getDestinationBlockPosFromOffset()` to not have [side effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science)), as it's not the type of function that I'd expect to have side effects)